### PR TITLE
Moving tests to individual processes. Coverage is broke and output is…

### PR DESCRIPTION
**Pull request by gauntface**
R: @jeffposnick @philipwalton 

Each test runs in it's own process now.

### CODE REVIEW

File: gulp-tasks/test.js [hunk 1]
```diff
@@ -1,47 +1,69 @@
...
-  runNodeTestsWithEnv('dev'),
+  () => runNodeTestsWithEnv('dev'),
 ));
 
+gulp.task('test:node:clean', () => {
+  return fse.remove(path.join(__dirname, '..', '.nyc_output'));
+});
+
+gulp.task('test:node:coverage', () => {
+  return spawn(getNpmCmd(), ['run', 'coverage-report']);
```
Comment (jeffposnick) [inline gulp-tasks/test.js]: When reading up on `gulp` v4, I learned that there is native support for using a `spawn`ed process as a return value from a task: https://www.joezimjs.com/javascript/complete-guide-upgrading-gulp-4/#4-Return-a-Child-Process

I think we need to keep `./utils/spawn-promise-wrapper` around for use cases where there are multiple subprocesses, or cleanup needs to be done after the subprocess runs, but it should be usable here.

File: gulp-tasks/test.js [hunk 2]
```diff
@@ -1,47 +1,69 @@
...
+    process.env.NODE_ENV = originalNodeEnv;
+
+    logHelper.error(err);
+    throw new Error(`[Workbox Error Msg] 'gulp test' discovered errors.`);
+  }
+};
+
+const runNodeTestsWithEnv = (nodeEnv) => {
+  const packagesToTest = glob.sync(`test/${global.packageOrStar}`);
+  return packagesToTest.reduce(async (promiseChain, packageToTest) => {
```
Comment (jeffposnick) [inline gulp-tasks/test.js]: One of the nicer parts of `async`/`await` [you don't have to use `reduce()` chains anymore](https://developers.google.com/web/fundamentals/primers/async-functions#example_outputting_fetches_in_order):

```js
const runNodeTestsWithEnv = async (nodeEnv) => {
  const packagesToTest = glob.sync(`test/${global.packageOrStar}`);
  for (const packageToTest of packagesToTest) {
    await runNodeTestSuite(packageToTest, nodeEnv);
  }
};
```

File: gulp-tasks/test.js (hunk 1 above)
Comment (gauntface) [reply gulp-tasks/test.js]: Good to know.

File: gulp-tasks/test.js [hunk 3]
```diff
@@ -1,47 +1,68 @@
...
-      options.push('--grep', global.cliOptions.grep);
-    }
-    const originalNodeEnv = process.env.NODE_ENV;
-
-    process.env.NODE_ENV = nodeEnv;
-    return spawn(getNpmCmd(), ['run', 'test', '--',
-      `./test/${global.packageOrStar}/node/**/*.{js,mjs}`,
+const runNodeTestSuite = async (testPath, nodeEnv) => {
+  logHelper.log(oneLine`
+  Running Node test on ${logHelper.highlight(testPath)}
```
Comment (philipwalton) [inline gulp-tasks/test.js]: The indentation looks messed up here.

**Pull request merged**